lib/yaml/constructor.py

```diff
@@ -125,11 +124,9 @@ def construct_mapping(self, node, deep=False):
         mapping = {}
         for key_node, value_node in node.value:
             key = self.construct_object(key_node, deep=deep)
-            try:
-                hash(key)
-            except TypeError, exc:
+            if not isinstance(key, collections.Hashable):
```

Comment:
This should be `collections.abc.Hashable` for Python 3 to support Python 3.8.  ref yaml/pyyaml#181